src/erlcloud_aws.erl

```diff
@@ -249,6 +250,111 @@ update_config(#aws_config{} = Config) ->
                    security_token = Credentials#metadata_credentials.security_token}}
     end.
 
+
+%%%---------------------------------------------------------------------------
+-spec service_config( Service :: atom() | string() | binary(),
+                      Region :: atom() | string() | binary(),
+                      Config :: #aws_config{} ) -> #aws_config{}.
+%%%---------------------------------------------------------------------------
+%% @doc Generate config updated to work with specified AWS service & region
+%%
+%% This function will generate a new configuration, based on the one
+%% provided, and configured to access the specified AWS service in the
+%% specified region.  If called for a service without a region based
+%% endpoint, the config provided will be returned unaltered.
+%%
+%% If an invalid service name is provided, then this will throw an error,
+%% presuming that this is just a coding error.  This behavior allows the
+%% chaining of calls to this interface to allow concise configuraiton of a
+%% config for multiple services.
+%%
+service_config( Service, Region, Config ) when is_list(Service) ->
+    service_config( list_to_binary(Service), Region, Config );
+service_config( Service, Region, Config ) when is_list(Region) ->
+    service_config( Service, list_to_binary(Region), Config );
+service_config( Service, Region, Config ) when is_atom(Service) ->
+    service_config( atom_to_binary(Service, latin1), Region, Config );
+service_config( Service, Region, Config ) when is_atom(Region) ->
+    service_config( Service, atom_to_binary(Region, latin1), Config );
```

Comment:
There's no need to use an atom as region :)
Please alter the guards order to: `is_atom(Service)`, `is_list(Service)`, `when is_list(Region)`